src/cogs/split.py

```diff
@@ -0,0 +1,205 @@
+    async def minimum_team_size(self, ctx: Context, min_team_size: int) -> None:
+        """Set the minimum players per team."""
+        if min_team_size <= 0:
+            await ctx.send("> Idk chief, this seems kinda weird. (*Please chose a number greater than 0*()", delete_after=5)
+            return
+
+        self.data.minimum_team_size = min_team_size
+        await ctx.send(f"> Minimum team size set to {min_team_size}", delete_after=5)
+
+    @config.command(aliases=["cap", "leader", "role"])
+    async def captain_role(self, ctx: Context, role: Role) -> None:
+        """Set the captain role."""
+        self.data.captain_role_id = role.id
+        await ctx.send(f"> Set captain role to {role.mention}", delete_after=5, allowed_mentions=AllowedMentions(roles=False))
+
+    @command()
+    async def split(self, ctx: Context) -> None:
+
+        # Get a list of invalid settings and display them to the user if there is
+        # at least one invalid setting before trying to split.
+        invalid_settings = []
+        if self.lobby_channel is None:
+            invalid_settings.append("Lobby")
+        if self.team_one_channel is None:
+            invalid_settings.append("Team One")
+        if self.team_two_channel is None:
+            invalid_settings.append("Team Two")
+
+        if len(invalid_settings) > 0:
+            await ctx.send(f"> You must configure the following settings before a split: {', '.join(invalid_settings)}", delete_after=5)
+            return
+
+        # FIXME: Find the underlying api request to get this info.
+        # This list is generated from the discord.py cache.
+        # This means if the user has been in the channel since the bot has come online,
+        # They will not be inside this list.
+        lobby_members = self.lobby_channel.members
+
+        if len(lobby_members) // 2 < self.data.minimum_team_size:
+            await ctx.send("> There is not enough members to meet the minimum team size", delete_after=5)
+            return
+
+        # Our randomization strategy is using random.shuffle then splitting the list in the middle
+        random.shuffle(lobby_members)
+
+        team_one, team_two = lobby_members[len(lobby_members) // 2:], lobby_members[:len(lobby_members) // 2]
+
+        embed = Embed(title="Teams", colour=EMBED_COLOR)
+        embed.add_field(name="__**Team 1**__", value='\n'.join([user.mention for user in team_one]))
+        embed.add_field(name="__**Team 2**__", value='\n'.join([user.mention for user in team_two]))
+        await ctx.send(embed=embed)
+
+        try:
+            for member in team_one:
+                await member.move_to(self.team_one_channel, reason="Team split")
+
+            for member in team_two:
+                await member.move_to(self.team_two_channel, reason="Team split")
+        except Forbidden:
+            await ctx.send("> It feels like I don't have the power to move people. Missing Permissions.")
+            return
```

Comment:
If a user leaves voice-chat before attempting to move them, you will get an `HTTPException` here.
```
discord.errors.HTTPException: 400 Bad Request (error code: 40032): Target user is not connected to voice.
```

Reply:
Here's what you could do instead:
```suggestion
        teams = (team_one, team_two)
        channels = (self.team_one_channel, self.team_two_channel)

        for team, channel in zip(teams, channels):
            for member in team:
                try:
                    await member.move_to(channel, reason="Team split")
                except Forbidden:
                    await ctx.send("> It feels like I don't have the power to move people. Missing Permissions.")
                    return
                except HTTPException as e:
                    if e.code == 40032: # target user not connected to voice
                        pass
                    else:
                        raise e
```

Reply:
The only issue with this is it will send the `Forbidden` error message for each user.

Reply:
Although I try to avoid high levels of indentation like this, This seems like the cleaner way to do this iteration. 
I'll commit it

Reply:
Ah yeah. The indentation bothered me too, but I couldn't see a way to avoid it.